backend/app/routes/UserController.js

```diff
@@ -229,14 +229,27 @@ router.put("/:id", async (req, res) => {
     if (queriedUser.isVolunteer === true)
       return res.status(400).send("User is already a volunteer\n");
 
+    if (req.body.certificateCode === undefined)
+      return res.status(400).send("Certificate code is required\n");
+
     const result = await User.updateOne(
       { _id: req.params.id },
-      { isVolunteer: true }
-    ).exec();
-
-    if (result.modifiedCount === 0)
-      return res.status(400).send("User is already a volunteer\n");
-
+      {
+        $set: {
+          "volunteer.isVolunteer": req.body.isVolunteer,
+          "volunteer.certificateCode": req.body.certificateCode,
+        },
+      }
+    );
+    if (result.modifiedCount === 0) {
+      if (req.body.isVolunteer === true) {
+        return res.status(400).send("User is already a volunteer");
+      } else {
```

Comment:
non va bene mettere else

Reply:
piuttosto metti una if in +, fuori dalla if che c'è già

Reply:
Non capisco cosa cambi all'atto pratico